aws/solutions/lambda-backed-cloudformation-custom-resources/get_vpc_main_route_table_id/Routetable.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import print_function
+import json
+import boto3
+import urllib
+from botocore.vendored import requests
+
+
+SUCCESS = "SUCCESS"
+FAILED = "FAILED"
+
+def send(event, context, responseStatus, responseData, physicalResourceId=None, noEcho=False):
+    responseUrl = event['ResponseURL']
+
+    print(responseUrl)
+
+    responseBody = {}
+    responseBody['Status'] = responseStatus
+    responseBody['Reason'] = 'See the details in CloudWatch Log Stream: ' + context.log_stream_name
+    responseBody['PhysicalResourceId'] = physicalResourceId or context.log_stream_name
+    responseBody['StackId'] = event['StackId']
+    responseBody['RequestId'] = event['RequestId']
+    responseBody['LogicalResourceId'] = event['LogicalResourceId']
+    responseBody['NoEcho'] = noEcho
+    responseBody['Data'] = responseData
+
+    json_responseBody = json.dumps(responseBody)
+    print("Response body:\n" + json_responseBody)
+    headers = {
+        'content-type' : '',
+        'content-length' : str(len(json_responseBody))
+    }
+
+    try:
+        response = requests.put(responseUrl,
+                                data=json_responseBody,
+                                headers=headers)
+        print("Status code: " + response.reason)
+    except Exception as e:
+        print("send(..) failed executing requests.put(..): " + str(e))
+
+
+print('Loading function')
+ec2 = boto3.client('ec2')
+
+def lambda_handler(event, context):
+    print("Received event: " + json.dumps(event, indent=2))
+    responseData={}
+
+    if(event['RequestType'] == 'Create'):
```

Comment:
Maybe worth pulling these out into functions to avoid length `if/else` blocks to aid readability.

Reply:
Added a function to improve readability